### PR TITLE
OSDOCS-2890: Updating for OAuth server audit logging

### PR DESCRIPTION
**Pull request by bergerhoffer**
Version(s):
4.11

Issue:
https://issues.redhat.com/browse/OSDOCS-2890

Link to docs preview:
* https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-policy-config.html#about-audit-log-profiles_audit-log-policy-config
* https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view
* https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#security-audit-log-basic-filtering_audit-log-view

Additional information:

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@ibihim Can you please review these doc updates and let me know if you have any feedback?

Previews:

* https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-policy-config.html#about-audit-log-profiles_audit-log-policy-config
* https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view
* https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#security-audit-log-basic-filtering_audit-log-view

CC @stlaz too

**Review by jeana-redhat — APPROVED** [review]
LGTM

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 1]
```diff
@@ -131,3 +131,43 @@ $ oc adm node-logs ci-ln-m0wpfjb-f76d1-vnb5x-master-0 --path=oauth-apiserver/aud
 ----
 {"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Metadata","auditID":"dd4c44e2-3ea1-4830-9ab7-c91a5f1388d6","stage":"ResponseComplete","requestURI":"/apis/user.openshift.io/v1/users/~","verb":"get","user":{"username":"system:serviceaccount:openshift-monitoring:prometheus-k8s","groups":["system:serviceaccounts","system:serviceaccounts:openshift-monitoring","system:authenticated"]},"sourceIPs":["10.0.32.4","10.128.0.1"],"userAgent":"dockerregistry/v0.0.0 (linux/amd64) kubernetes/$Format","objectRef":{"resource":"users","name":"~","apiGroup":"user.openshift.io","apiVersion":"v1"},"responseStatus":{"metadata":{},"code":200},"requestReceivedTimestamp":"2021-03-08T17:47:43.653187Z","stageTimestamp":"2021-03-08T17:47:43.660187Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":"RBAC: allowed by ClusterRoleBinding \"basic-users\" of ClusterRole \"basic-user\" to Group \"system:authenticated\""}}
 ----
+
+* View the OpenShift OAuth server logs:
```
Comment (ibihim) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:135]: Does it make sense to emphasize, that those are audit logs and not the defaults logs one would get by calling `kubectl logs $POD`?

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 2]
```diff
@@ -131,3 +131,43 @@ $ oc adm node-logs ci-ln-m0wpfjb-f76d1-vnb5x-master-0 --path=oauth-apiserver/aud
...
+
+.. List the OpenShift OAuth server logs that are available for each control plane node:
```
Comment (ibihim) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:137]: Same as above.

**Comment by bergerhoffer** [comment]
@ibihim I made a few updates per the feedback. Can you please re-review?

Updated preview link is now here: http://file.rdu.redhat.com/~ahoffer/2022/OSDOCS-2890/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view

**Comment by bergerhoffer** [comment]
@xingxingxia Can you please QE review when you get a chance?

Preview: http://file.rdu.redhat.com/~ahoffer/2022/OSDOCS-2890/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view

File: modules/nodes-nodes-audit-config-about.adoc [hunk 3]
```diff
@@ -6,7 +6,7 @@
 [id="about-audit-log-profiles_{context}"]
 = About audit log policy profiles
 
-Audit log profiles define how to log requests that come to the OpenShift API server, the Kubernetes API server, and the OAuth API server.
+Audit log profiles define how to log requests that come to the OpenShift API server, Kubernetes API server, OpenShift OAuth API server, and OpenShift OAuth server.
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-config-about.adoc:9]: I tested: WriteRequestBodies and AllRequestBodies profiles don't change OpenShift OAuth server's auditing behavior from "Default", "None" profile disables OpenShift OAuth server's auditing. The wording "define how to log requests that come" in the following table may not be accurate for WriteRequestBodies and AllRequestBodies profiles for OpenShift OAuth server.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 4]
```diff
@@ -131,3 +131,43 @@ $ oc adm node-logs ci-ln-m0wpfjb-f76d1-vnb5x-master-0 --path=oauth-apiserver/aud
...
+----
+ci-ln-m0wpfjb-f76d1-vnb5x-master-0 audit-2022-05-11T18-57-32.395.log
+ci-ln-m0wpfjb-f76d1-vnb5x-master-0 audit.log
+ci-ln-m0wpfjb-f76d1-vnb5x-master-1 audit-2022-05-11T19-07-07.021.log
+ci-ln-m0wpfjb-f76d1-vnb5x-master-1 audit.log
+ci-ln-m0wpfjb-f76d1-vnb5x-master-2 audit-2022-05-11T19-06-51.844.log
+ci-ln-m0wpfjb-f76d1-vnb5x-master-2 audit.log
+----
+
+.. View a specific OpenShift OAuth server log by providing the node name and the log name:
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:155]: > View a specific OpenShift OAuth server log
View a specific OpenShift OAuth server audit log

Same for other places "View a specific OpenShift API server log", "View a specific Kubernetes API server log", "View a specific OpenShift OAuth API server log"

File: modules/nodes-nodes-audit-config-about.adoc (hunk 3 above)
Comment (xingxingxia) [reply modules/nodes-nodes-audit-config-about.adoc:9]: Ah, OK, I see there is a "`.small`" note for it, then no issue about the wording.

**Comment by xingxingxia** [comment]
```
Gathering audit logs
Run the oc adm must-gather command with the -- /usr/bin/gather_audit_logs flag
```
/usr/bin/gather_audit_logs is not a "flag". It is a command, this can seen from oc adm must-gather -h:
```
$ oc adm must-gather -h
  # Gather information using a specific image, command, and pod-dir
  oc adm must-gather --image=my/image:tag --source-dir=/pod/directory -- myspecial-command.sh
```

**Comment by xingxingxia** [comment]
Need we somewhere mention "authentication.openshift.io/decision" can be one of allow, deny, error?

**Comment by bergerhoffer** [comment]
> In "Gathering audit logs" section (not shown in this PR "Files changed"; happened to find it in your preview link)
> 
> ```
> Run the oc adm must-gather command with the -- /usr/bin/gather_audit_logs flag
> ```
> 
> /usr/bin/gather_audit_logs is not a "flag". It is a command, this can seen from oc adm must-gather -h:
> 
> ```
> $ oc adm must-gather -h
>   # Gather information using a specific image, command, and pod-dir
>   oc adm must-gather --image=my/image:tag --source-dir=/pod/directory -- myspecial-command.sh
> ```

Thanks @xingxingxia - I'm actually not familiar with this file so I'll track it down and will update to use the correct phrasing.

**Comment by bergerhoffer** [comment]
> Need we somewhere mention "authentication.openshift.io/decision" can be one of allow, deny, error?

@xingxingxia I don't really see anywhere else where we really get into the possible values for fields of an audit event, outside of generically mentioning the fields in this table: http://file.rdu.redhat.com/~ahoffer/2022/OSDOCS-2890/security/audit-log-view.html#nodes-pods-audit-log-basic_audit-log-view.

So I don't _think_ so, but let me know if you feel strongly about it and we can look into it more.

**Comment by xingxingxia** [comment]
AUTH-6 OEP https://github.com/openshift/enhancements/blob/master/enhancements/authentication/login-logout-events.md#proposal uses quite length to tell allow, deny or error. For AUTH-6, this is a feature point specific and non-trivial. So I prefer to mention allow, deny or error if we documen AUTH-6.

**Comment by bergerhoffer** [comment]
> [AUTH-6](https://issues.redhat.com//browse/AUTH-6) OEP https://github.com/openshift/enhancements/blob/master/enhancements/authentication/login-logout-events.md#proposal uses quite length to tell allow, deny or error. For [AUTH-6](https://issues.redhat.com//browse/AUTH-6), this is a feature point specific and non-trivial. So I prefer to mention allow, deny or error if we documen [AUTH-6](https://issues.redhat.com//browse/AUTH-6).

@xingxingxia Sure thing then! I mentioned this in the part about viewing the audit logs. Can you take a look and let me know if this works? Thanks!

http://file.rdu.redhat.com/~ahoffer/2022/OSDOCS-2890/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view

**Comment by xingxingxia** [comment]
LGTM, but the preview link https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view page does not show "The possible values for the authentication.openshift.io/decision annotation are allow, deny, or error.", what is wrong?

**Comment by bergerhoffer** [comment]
> LGTM, but the preview link https://deploy-preview-45799--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view page does not show "The possible values for the authentication.openshift.io/decision annotation are allow, deny, or error.", what is wrong?

@xingxingxia Apolgoies - our automatic preview system has not been functioning for a month or two, so we've been having to upload manual previews instead. As long as you see the proper text on the latest preview link I shared (http://file.rdu.redhat.com/~ahoffer/2022/OSDOCS-2890/security/audit-log-view.html#nodes-nodes-audit-log-basic-viewing_audit-log-view), we should be good to go!

**Comment by xingxingxia** [comment]
Ah, thanks for the explanation!
/lgtm

**Pull request merged**